Client/Frontend/Browser/TabManager.swift

```diff
@@ -483,8 +483,9 @@ extension TabManager {
             if let screenshotUUID = savedTab.screenshotUUID,
                let imageStore = self.imageStore {
                 tab.screenshotUUID = screenshotUUID
-                imageStore.get(screenshotUUID.UUIDString) >>== { screenshot in
-                    if tab.screenshotUUID == screenshotUUID {
+                imageStore.get(screenshotUUID.UUIDString).uponQueue(dispatch_get_main_queue()) { result in
+                    if let data = result.successValue where tab.screenshotUUID == screenshotUUID {
+                        let screenshot = UIImage(data: data)
                         tab.setScreenshot(screenshot, revUUID: false)
```

Comment:
This line is the only one that needs to be run on the main thread, so I think you should roll back the `UIImage` to `NSData` change and see if this still fixes the bug.